consensus/state.go

```diff
@@ -1100,6 +1103,19 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 	} else if !cs.replayMode {
 		cs.Logger.Error("enterPropose: Error signing proposal", "height", height, "round", round, "err", err)
 	}
+
+	// post data to ipfs
+	// TODO(evan): don't hard code context and timeout
+	if cs.IpfsAPI != nil {
+		// longer timeouts result in block proposers failing to propose blocks in time.
+		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*1500)
```

Comment:
Nice, I didn't know that. In this case, we need the timeout because `PutBlock` is synchronous and will take too long, stopping the block proposer from proposing a block in time.